pupil_src/shared_modules/video_capture/utils.py

```diff
@@ -211,10 +210,13 @@ def load_ts(self):
             return
         self.ts = self._fix_negative_time_jumps(self.ts)
 
-    def load_pts(self):
-        packets = self.cont.demux(video=0)
+    def load_pts(self, container=None):
+        if container is None:
+            container = self.attempt_open_container()
```

Comment:
This function is actually only called, if the container is valid: https://github.com/pupil-labs/pupil/pull/1835/files#diff-84845973e9fa00c663d23ca5d1efa86cR358

Nonetheless, we should not be leaking the `AVError`. Instead I would raise `InvalidContainerError` as suggested below.

Reply:
I think you could mitigate this by always requiring a container.
Then the container has to be valid when passing this in.
Since this function seems to be called only once in a context where we pass a valid container, correct?

Reply:
agreed